src/firebolt/async_db/util.py

```diff
@@ -12,14 +12,19 @@
     InterfaceError,
 )
 from firebolt.utils.urls import GATEWAY_HOST_BY_ACCOUNT_NAME
-from firebolt.utils.util import parse_url_and_params
+from firebolt.utils.util import UtilCache, parse_url_and_params
+
+_firebolt_async_system_engine_cache = UtilCache[Tuple[str, Dict[str, str]]]()
 
 
 async def _get_system_engine_url_and_params(
     auth: Auth,
     account_name: str,
     api_endpoint: str,
 ) -> Tuple[str, Dict[str, str]]:
+    cache_key = f"{account_name}-{api_endpoint}"
```

Comment:
We still create a key manually here. But I was thinking even more, just for get to have a signature `def get(*key):` and then inside do `cache_key = self.create_key(*key)` or something simillar

Reply:
Whoops, missed this one. I see where you're getting at now. My only problem with this is `def set(*key, value)` can lead to non-obvious errors when value is forgotten from the call, e.g. `cache.set(account_name, api_endpoint)` will work just fine, but lead to runtime issues. I've pushed a different approach with key accepting anything and using __repr__ to convert to a hashable key. Hopefully it solves both issues.